src/dmd/staticcond.d

```diff
@@ -27,6 +27,12 @@ import dmd.tokens;
 
 
 
+struct FailedExpression
```

Comment:
Not a huge fan of this name. `ConstraintFailResult`? The existing name doesn't convey what the type represents (i.e. it has a message too) and seems like a subclass of `Expression` at a glance.